engine/src/main/java/org/terasology/rendering/dag/DefaultStateChangeStorage.java

```diff
@@ -15,20 +15,19 @@
  */
 package org.terasology.rendering.dag;
 
-import java.util.List;
+import java.util.HashMap;
+import static org.lwjgl.opengl.GL11.GL_FILL;
 
 /**
- *
+ * TODO: Add javadocs
  */
-public class RenderPipelineExecutor {
-
-    private List<PipelineTask> taskList;
+class DefaultStateChangeStorage extends HashMap<Class<? extends StateChange>, StateChange> {
```

Comment:
I'm sorry, I really don't like this central storage solution.

What I would like to see is this:

class SetColorStateChange implements StateChange {

    private static final defaultInstance;
    private final Color value;

   SetColorStateChange(Color aColor) {
        value = aColor;
        defaultInstance = SetColorStateChange(hardcodedDefaultValue)
    }

   public SetColorStateChange getDefaultInstance() {
       return defaultInstance;
   }

   public Color getValue() {
       return value;
   } 
}

This way we keep all the information regarding a StateChange in one place.